### ISSUE TITLE
The result is too noisy

### ISSUE THREAD

**Issue by timerobin** [Janspiry/Image-Super-Resolution-via-Iterative-Refinement#78]
Thank you for the code.
I have trained the sr3 model on the images of different resolution, like 16->128, 64->512, 256->1024 on ffhq and celebahq.
The iters are 50k, and the learning rate is 3e-6.
But I found the val result are with too much noise, with n_timestep=2000.
This is the result of 512*512
![50000_3_sr](https://user-images.githubusercontent.com/26972401/189134591-681d0113-3171-4482-979f-c1c8fc538ee9.png)
And the loss function is not stable convergence

**Comment by Janspiry** [comment]
![image](https://user-images.githubusercontent.com/32922494/189822312-25af4c8b-1df2-459f-9772-da86b98e6a02.png)
Yes, something similar happened to me at the time when I was training, and I guess the model's learning ability was limited. But I didn't have a better GPU resources, so I didn't bother to optimize it.

**Comment by timerobin** [comment]
> 

Thank you very much for your reply. 
I have ran more experiments and I find image noise disappears when more resblocks are used.
But the phenomenon of hue deviation still exists.
Could you please give some advice on how to tune hyperparameters to remove color bias

**Comment by Janspiry** [comment]
I have two guesses for now, but not sure about the specifics:
1. the set up of group convolution that interact between channels.
2. the setting of clip noise in [https://github.com/Janspiry/Image-Super-Resolution-via-Iterative-Refinement/blob/ ef9b943b573328d7a5ddb1a0c2abd168b91610dc/model/sr3_modules/diffusion.py#L162](https://github.com/Janspiry/Image-Super-Resolution-via-Iterative-Refinement/blob/ef9b943b573328d7a5ddb1a0c2abd168b91610dc/model/sr3_modules/diffusion.py#L162)

**Comment by KANGXI123456** [comment]
> 

Excuse me, can you tell me where to add the resblocks?
Thanks you very much

**Comment by ElliotQi** [comment]
> > 
> 
> Excuse me, can you tell me where to add the resblocks? Thanks you very much

An easy way is to modify the number of model.unet.resblock  in the config file. But I still get noisy image :<

**Comment by 294coder** [comment]
@KANGXI123456 
why `x_recon` is clipped in `(-1, 1)`, it can help remove color bias?

**Comment by YangMei66** [comment]
> > 
> 
> Thank you very much for your reply. I have ran more experiments and I find image noise disappears when more resblocks are used. But the phenomenon of hue deviation still exists. Could you please give some advice on how to tune hyperparameters to remove color bias

Hello, I confront the same issue about hue deviation. How did you solve the issue?
![image](https://user-images.githubusercontent.com/63027996/232272620-193b7f9c-df3f-473b-ac0e-bbe9ec7a1e42.png)

**Comment by 294coder** [comment]
@YangMei66 hi, you may find some solutions here: #69

**Comment by YangMei66** [comment]
This is an automatic reply, confirming that your e-mail was received. Thank you.--------------------------------------------Best,Mei Yang

**Comment by YangMei66** [comment]
> @YangMei66 hi, you may find some solutions here: #69

thank you
